src/web/releases.rs

```diff
@@ -676,3 +692,96 @@ pub fn build_queue_handler(req: &mut Request) -> IronResult<Response> {
         .set_true("releases_queue_tab")
         .to_resp("releases_queue")
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::test::{wrapper, TestDatabase};
+
+    #[test]
+    fn database_search() {
+        wrapper(|env| {
+            let db = env.db();
+
+            db.fake_release().name("foo").version("0.0.0").create()?;
+            db.fake_release()
+                .name("bar-foo")
+                .version("0.0.0")
+                .create()?;
+            db.fake_release()
+                .name("foo-bar")
+                .version("0.0.1")
+                .create()?;
+            db.fake_release().name("fo0").version("0.0.0").create()?;
+            db.fake_release()
+                .name("fool")
+                .version("0.0.0")
+                .build_result_successful(false)
+                .create()?;
+            db.fake_release()
+                .name("freakin")
+                .version("0.0.0")
+                .create()?;
+            db.fake_release()
+                .name("something unreleated")
+                .version("0.0.0")
+                .create()?;
+
+            let (num_results, results) = get_search_results(&db.conn(), "foo", 1, 100);
+            let mut results = results.into_iter();
+
+            assert_eq!(num_results, 4);
+
+            let expected = ["foo", "fo0", "bar-foo", "foo-bar"];
+            for expected in expected.iter() {
+                assert_eq!(expected, &results.next().unwrap().name);
+            }
+            assert!(results.collect::<Vec<_>>().is_empty());
+
+            Ok(())
+        })
+    }
+
+    fn non_exact(db: &TestDatabase) -> Result<(), crate::error::Error> {
+        db.fake_release().name("reg3x").version("0.0.0").create()?;
+        db.fake_release().name("regex-").version("0.0.0").create()?;
+        db.fake_release()
+            .name("regex-syntax")
+            .version("0.0.0")
+            .create()?;
+
+        Ok(())
+    }
+
+    fn rest_non_exact(mut rest: Vec<Release>) {
+        for name in ["reg3x", "regex-", "regex-syntax"].iter() {
+            assert_eq!(rest.remove(0).name, *name);
+        }
+
+        assert!(rest.is_empty());
+    }
+
+    #[test]
+    fn exacts_dont_care() {
+        let near_matches = ["Regex", "rEgex", "reGex", "regEx", "regeX"];
+
+        for name in near_matches.iter() {
+            wrapper(|env| {
+                let db = env.db();
+                db.fake_release().name(name).version("0.0.0").create()?;
+
+                non_exact(&db)?;
+
+                let (num_results, results) = get_search_results(&db.conn(), "foo", 1, 100);
```

Comment:
I don't understand what this is testing. Shouldn't there be no results for "foo" because you haven't released any? Or does this depend on `database_search` running first?

Reply:
That is a typo, my bad